File: ui_lib/src/main/java/com/bride/ui_lib/BaseFragment.java
package com.bride.ui_lib;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;

import androidx.fragment.app.Fragment;

import timber.log.Timber;

/**
 * <p>Created by shixin on 2019/3/22.
 */
public class BaseFragment extends Fragment {
    protected static String TAG;
    private final Handler mHandler = new Handler(Looper.getMainLooper());

    {
        TAG = getClass().getSimpleName();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Timber.tag(TAG).d("onCreate");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Timber.tag(TAG).d("onDestroy");
    }

    protected Handler getHandler() {
        return mHandler;
    }
}

File: ui_lib/src/main/java/com/bride/ui_lib/BaseFragment.kt
package com.bride.ui_lib

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.fragment.app.Fragment
import timber.log.Timber

open class BaseFragment : Fragment() {
    companion object {
        @JvmStatic
        protected lateinit var TAG: String
    }

    protected val handler = Handler(Looper.getMainLooper())

    init {
        TAG = javaClass.simpleName
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Timber.tag(TAG).d("onCreate")
    }

    override fun onDestroy() {
        super.onDestroy()
        Timber.tag(TAG).d("onDestroy")
    }
}